### DISCUSSION TITLE
Is there anyone here that working with Swift by using Neovim?

### DISCUSSION THREAD

**Discussion by bengidev** [neovim/neovim#27073] [General]
I've been struggling to setup Neovim configuration to be able to work with Swift. If anyone in here work with Swift, please help me to setup my Neovim config for Swift.

This is my repo for my current NeoVim config:
https://github.com/bengidev/Bengi_LazyVim_Configuration

 Thanks for your help 🍻

**Comment by andreacfromtheapp** [comment]
hi, give this a try https://github.com/wojciech-kulik/xcodebuild.nvim

**Reply by bengidev** [reply]
Yep, i've already give it a try and try to copy/paste the codes. But when i open my Swift file, there's nothing happen. My lsp not trigger to start. You can check my config from my link above.

**Reply by andreacfromtheapp** [reply]
I tried and failed myself, sticking with XCode while I'm learning. Will try again the future. However, the repo has a nimv starter template link as well. Check it out. Good luck!

**Comment by martinmose** [comment]
@bengidev Hello,

I've just started using Neovim and came across your setup. It looks really nice! I got some great inspiration from your repository, so thanks a lot for that 🥳

It seems like I don't have anything you don't, but overall, everything is working great 🤩

**Comment by erlangparasu** [comment]
https://www.swift.org/documentation/articles/zero-to-swift-nvim.html

**Comment by filip-sakel** [comment]
Here's what worked for my [LazyVim](https://www.lazyvim.org/) setup:
```lua
-- lua/plugins/sourcekit-lsp.lua
return {
    {
        "neovim/nvim-lspconfig",
        opts = {
            servers = {
                swift = {
                    cmd = {
                        "sourcekit-lsp",
                    },
                    filetypes = { "swift" },
                    root_markers = {
                        ".git",
                        "compile_commands.json",
                        ".sourcekit-lsp",
                        "Package.swift",
                    },
                    get_language_id = function(_, ftype)
                        return ftype
                    end,
                    capabilities = {
                        workspace = {
                            didChangeWatchedFiles = {
                                dynamicRegistration = true,
                            },
                        },
                        textDocument = {
                            diagnostic = {
                                dynamicRegistration = true,
                                relatedDocumentSupport = true,
                            },
                        },
                    },
                },
            },
        },
    },
}
```
Based on this [discussion](https://github.com/LazyVim/LazyVim/discussions/2263#discussioncomment-8223963) and the Swift blog [article](https://www.swift.org/documentation/articles/zero-to-swift-nvim.html) linked above. You should create a lua file in the plugins folder of your LazyVim config, in my case: `~/.config/nvim/lua/plugins/sourcekit-lsp.lua`. The default LazyVim config should automatically pick up this new plugin file, and the built-in Mason and blink-cmp plugins should work with the SourceKit LSP server. 

(For reference, I'm using LazyVim version 8 and NeoVim 0.12.2)
